diff --git a/xbmc/interfaces/json-rpc/AudioLibrary.cpp b/xbmc/interfaces/json-rpc/AudioLibrary.cpp
index ab08cdf833..bed043b421 100644
--- a/xbmc/interfaces/json-rpc/AudioLibrary.cpp
+++ b/xbmc/interfaces/json-rpc/AudioLibrary.cpp
@@ -32,6 +32,8 @@
 
 #include <memory>
 
+#include <music/MusicLibraryQueue.h>
+
 using namespace MUSIC_INFO;
 using namespace JSONRPC;
 using namespace XFILE;
@@ -1354,6 +1356,60 @@ JSONRPC_STATUS CAudioLibrary::GetAdditionalSongDetails(const CVariant& parameter
   return OK;
 }
 
+JSONRPC_STATUS CAudioLibrary::RefreshArtist(const std::string& method,
+                                            ITransportLayer* transport,
+                                            IClient* client,
+                                            const CVariant& parameterObject,
+                                            CVariant& result)
+{
+  int artistID = (int)parameterObject["artistid"].asInteger();
+
+  CMusicDbUrl musicUrl;
+  if (!musicUrl.FromString("musicdb://artists/"))
+    return InternalError;
+
+  CMusicDatabase musicdatabase;
+  if (!musicdatabase.Open())
+    return InternalError;
+
+  //checking if artistID is a valid one
+  if (!musicdatabase.GetArtistExists(artistID))
+    return InvalidParams;
+
+  //set the artist id on the musicdb url
+  musicUrl.AddOption("artistid", artistID);
+
+  //executing the StartArtistScan for refreshing the artist scraped informations
+  CMusicLibraryQueue::GetInstance().StartArtistScan(musicUrl.ToString(), true);
+
+  return ACK;
+}
+
+JSONRPC_STATUS CAudioLibrary::RefreshAlbum(const std::string& method,
+                                           ITransportLayer* transport,
+                                           IClient* client,
+                                           const CVariant& parameterObject,
+                                           CVariant& result)
+{
+  int albumID = (int)parameterObject["albumid"].asInteger();
+
+  CMusicDatabase musicdatabase;
+  if (!musicdatabase.Open())
+    return InternalError;
+
+  //check if albumID is a valid one
+  CAlbum album;
+  if (!musicdatabase.GetAlbum(albumID, album, false))
+    return InvalidParams;
+
+  std::string path = StringUtils::Format("musicdb://albums/{}/", albumID);
+
+  //execute the album refresh job
+  CMusicLibraryQueue::GetInstance().StartAlbumScan(path, true);
+
+  return ACK;
+}
+
 bool CAudioLibrary::CheckForAdditionalProperties(const CVariant &properties, const std::set<std::string> &checkProperties, std::set<std::string> &foundProperties)
 {
   if (!properties.isArray() || properties.empty())
diff --git a/xbmc/interfaces/json-rpc/AudioLibrary.h b/xbmc/interfaces/json-rpc/AudioLibrary.h
index 9946f9d998..674bb0a0d3 100644
--- a/xbmc/interfaces/json-rpc/AudioLibrary.h
+++ b/xbmc/interfaces/json-rpc/AudioLibrary.h
@@ -69,6 +69,16 @@ namespace JSONRPC
     static JSONRPC_STATUS GetAdditionalSongDetails(const CVariant& parameterObject,
                                                    const CFileItemList& items,
                                                    CMusicDatabase& musicdatabase);
+    static JSONRPC_STATUS RefreshArtist(const std::string& method,
+                                        ITransportLayer* transport,
+                                        IClient* client,
+                                        const CVariant& parameterObject,
+                                        CVariant& result);
+    static JSONRPC_STATUS RefreshAlbum(const std::string& method,
+                                       ITransportLayer* transport,
+                                       IClient* client,
+                                       const CVariant& parameterObject,
+                                       CVariant& result);
 
   private:
     static void FillAlbumItem(const CAlbum& album,
diff --git a/xbmc/interfaces/json-rpc/JSONServiceDescription.cpp b/xbmc/interfaces/json-rpc/JSONServiceDescription.cpp
index 0ce49b9898..a5b918f64f 100644
--- a/xbmc/interfaces/json-rpc/JSONServiceDescription.cpp
+++ b/xbmc/interfaces/json-rpc/JSONServiceDescription.cpp
@@ -119,6 +119,8 @@ JsonRpcMethodMap CJSONServiceDescription::m_methodMaps[] = {
   { "AudioLibrary.SetArtistDetails",                CAudioLibrary::SetArtistDetails },
   { "AudioLibrary.SetAlbumDetails",                 CAudioLibrary::SetAlbumDetails },
   { "AudioLibrary.SetSongDetails",                  CAudioLibrary::SetSongDetails },
+  { "AudioLibrary.RefreshArtist",                   CAudioLibrary::RefreshArtist },
+  { "AudioLibrary.RefreshAlbum",                    CAudioLibrary::RefreshAlbum },
   { "AudioLibrary.Scan",                            CAudioLibrary::Scan },
   { "AudioLibrary.Export",                          CAudioLibrary::Export },
   { "AudioLibrary.Clean",                           CAudioLibrary::Clean },
diff --git a/xbmc/interfaces/json-rpc/schema/methods.json b/xbmc/interfaces/json-rpc/schema/methods.json
index 19be352bb1..aa70a2879f 100644
--- a/xbmc/interfaces/json-rpc/schema/methods.json
+++ b/xbmc/interfaces/json-rpc/schema/methods.json
@@ -3063,6 +3063,34 @@
     ],
     "returns": "string"
   },
+  "AudioLibrary.RefreshArtist": {
+    "type": "method",
+    "description": "Refresh the given artist in the library",
+    "transport": "Response",
+    "permission": "UpdateData",
+    "params": [
+      {
+        "name": "artistid",
+        "$ref": "Library.Id",
+        "required": true
+      }
+    ],
+    "returns": "string"
+  },
+  "AudioLibrary.RefreshAlbum": {
+    "type": "method",
+    "description": "Refresh the given album in the library",
+    "transport": "Response",
+    "permission": "UpdateData",
+    "params": [
+      {
+        "name": "albumid",
+        "$ref": "Library.Id",
+        "required": true
+      }
+    ],
+    "returns": "string"
+  },
   "AudioLibrary.Scan": {
     "type": "method",
     "description": "Scans the audio sources for new library items",
diff --git a/xbmc/interfaces/json-rpc/schema/version.txt b/xbmc/interfaces/json-rpc/schema/version.txt
index 4727db3092..eecf12c86d 100644
--- a/xbmc/interfaces/json-rpc/schema/version.txt
+++ b/xbmc/interfaces/json-rpc/schema/version.txt
@@ -1 +1 @@
-JSONRPC_VERSION 13.6.0
+JSONRPC_VERSION 13.7.0
